arrow/extensions/geoarrow_point.go

```diff
@@ -0,0 +1,407 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"reflect"
+	"strings"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+)
+
+// Point represents a point geometry with coordinates
+type Point struct {
+	X, Y, Z, M float64
+	Dimension  CoordinateDimension
+}
+
+// NewPoint creates a new 2D point
+func NewPoint(x, y float64) Point {
+	return Point{X: x, Y: y, Dimension: DimensionXY}
+}
+
+// NewPointZ creates a new 3D point with Z coordinate
+func NewPointZ(x, y, z float64) Point {
+	return Point{X: x, Y: y, Z: z, Dimension: DimensionXYZ}
+}
+
+// NewPointM creates a new 2D point with M coordinate
+func NewPointM(x, y, m float64) Point {
+	return Point{X: x, Y: y, M: m, Dimension: DimensionXYM}
+}
+
+// NewPointZM creates a new 3D point with Z and M coordinates
+func NewPointZM(x, y, z, m float64) Point {
+	return Point{X: x, Y: y, Z: z, M: m, Dimension: DimensionXYZM}
+}
+
+// NewEmptyPoint creates a new empty point
+func NewEmptyPoint() Point {
```

Comment:
The comment for `IsEmpty` says that an empty point would have no dimension, so we shouldn't initialize it with `DimensionXY`